crates/sui-indexer-alt-framework/src/pipeline/concurrent/commit_watermark.rs

```diff
@@ -21,6 +21,8 @@ use crate::{
     store::{Connection, Store},
 };
 
+#[cfg(test)]
```

Comment:
ditto on the above feedback -- this pattern should be very rare -- only required if you need to add a test-only function to the implementation. I'll leave off commenting on the rest!